node/modules/core.go

```diff
@@ -69,45 +70,71 @@ func MemoryConstraints() system.MemoryConstraints {
 
 // MemoryWatchdog starts the memory watchdog, applying the computed resource
 // constraints.
-func MemoryWatchdog(lc fx.Lifecycle, constraints system.MemoryConstraints) {
+func MemoryWatchdog(lr repo.LockedRepo, lc fx.Lifecycle, constraints system.MemoryConstraints) {
 	if os.Getenv(EnvWatchdogDisabled) == "1" {
 		log.Infof("memory watchdog is disabled via %s", EnvWatchdogDisabled)
 		return
 	}
 
-	cfg := watchdog.MemConfig{
-		Resolution: 5 * time.Second,
-		Policy: &watchdog.WatermarkPolicy{
-			Watermarks:         []float64{0.50, 0.60, 0.70, 0.85, 0.90, 0.925, 0.95},
-			EmergencyWatermark: 0.95,
-		},
-		Logger: logWatchdog,
+	// configure heap profile capture so that one is captured per episode where
+	// utilization climbs over 90% of the limit. A maximum of 10 heapdumps
+	// will be captured during life of this process.
+	watchdog.HeapProfileDir = filepath.Join(lr.Path(), "heapprof")
+	watchdog.HeapProfileMaxCaptures = 10
+	watchdog.HeapProfileThreshold = 0.1
```

Comment:
Oops, this value leaked from testing. It should be 0.9. Will follow up with a PR.